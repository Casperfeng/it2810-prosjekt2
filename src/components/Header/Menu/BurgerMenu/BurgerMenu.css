.burgermenu {
  z-index: 100;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.bar {
  transition: 0.3s;
  margin-top: 2.5px;
  margin-bottom: 2.5px;
  background-color: white;
  width: 40px;
  height: 5px;
}

.event1 {
  transform: rotate(-45deg) translate(-6px, 8px);
}

.event2 {
  opacity: 0;
}

.event3 {
  transform: rotate(45deg) translate(-6px, -8px);
}
