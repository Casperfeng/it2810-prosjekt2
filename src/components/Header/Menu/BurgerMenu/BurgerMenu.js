import React from 'react';
import './BurgerMenu.css';

export default function BurgerMenu(props) {
  //brukes til Ã¥ 'triggere' burgermeny animasjonen

  const triggerAnimation = props.clicked ? 'event' : '';
  return (
    <div className='burgermenu'>
      <div className={`bar ${triggerAnimation}1`} />
      <div className={`bar ${triggerAnimation}2`} />
      <div className={`bar ${triggerAnimation}3`} />
    </div>
  );
}
