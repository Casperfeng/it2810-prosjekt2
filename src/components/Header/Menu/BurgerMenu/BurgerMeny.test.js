import React from 'react';
import BurgerMenu from './BurgerMenu';
import renderer from 'react-test-renderer';

it('renders correctly when unclicked', () => {
  const tree = renderer
    .create(<BurgerMenu clicked={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when clicked', () => {
  const tree = renderer
    .create(<BurgerMenu clicked={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});



