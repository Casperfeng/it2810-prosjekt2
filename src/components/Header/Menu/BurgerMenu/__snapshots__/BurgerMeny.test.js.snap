// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when clicked 1`] = `
<div
  className="burgermenu"
>
  <div
    className="bar event1"
  />
  <div
    className="bar event2"
  />
  <div
    className="bar event3"
  />
</div>
`;

exports[`renders correctly when unclicked 1`] = `
<div
  className="burgermenu"
>
  <div
    className="bar 1"
  />
  <div
    className="bar 2"
  />
  <div
    className="bar 3"
  />
</div>
`;
